rootfs/apply.go

```diff
@@ -69,7 +69,7 @@ func ApplyLayer(snapshots snapshot.Snapshotter, mounter Mounter, rd io.Reader, p
 	digester := digest.Canonical.Digester() // used to calculate diffID.
 	rd = io.TeeReader(rd, digester.Hash())
 
-	if _, err := archive.Apply(context.Background(), key, rd); err != nil {
+	if _, err := archive.Apply(context.Background(), dir, rd); err != nil {
```

Comment:
Please, let's read the [comment above](https://github.com/containerd/containerd/pull/755/files#diff-6ea9abd57ac207460460749d57c0b8f9R45). `dir` and `key` are identical. The problem here is that key isn't selected appropriately.

Yes, this should be `dir`, but the reason we run into extra unpacking is because `key` is simply random. A careful selection of `key` is required to address the original concern of this PR.